File: src/01-easy/the-river-ii-/solution.scala
object Solution extends App {
  def ds(n:Long):Long={var x=n;var s=0L;while(x>0){s+=x%10;x/=10};s}
  val r=scala.io.StdIn.readLong()
  println(if((math.max(1L,r-45) until r).exists(x=>x+ds(x)==r))"YES" else "NO")
}

File: src/01-easy/the-river-ii-/solution.dart
import 'dart:io';
import 'dart:math';
int ds(int n){int s=0;while(n>0){s+=n%10;n=n~/10;}return s;}
void main(){
  int r=int.parse(stdin.readLineSync()!);
  for(int x=max(1,r-45);x<r;x++)if(x+ds(x)==r){print("YES");return;}
  print("NO");
}
